src/test/auth/aws.rs

```diff
@@ -1,125 +1,130 @@
-
-    let client = get_client().await;
-    let coll = client.database("aws").collection::<Document>("somecoll");
-    match coll.find_one(doc! {}).await {
-        Ok(_) => panic!(
-            "find one should have failed with authentication error due to poisoned cached \
-             credential"
-        ),
-        Err(error) => assert!(error.is_auth_error()),
-    }
-    assert!(cached_credential().await.is_none());
-
-    coll.find_one(doc! {}).await.unwrap();
-    assert!(cached_credential().await.is_some());
-}
-
-#[tokio::test]
-async fn credential_caching_environment_vars() {
-    // This test should only be run when authenticating using AWS endpoints.
-    if var("SKIP_CREDENTIAL_CACHING_TESTS").is_ok() {
-        return;
-    }
-
-    clear_cached_credential().await;
-
-    let client = get_client().await;
-    let coll = client.database("aws").collection::<Document>("somecoll");
-    coll.find_one(doc! {}).await.unwrap();
-    assert!(cached_credential().await.is_some());
-
-    set_var("AWS_ACCESS_KEY_ID", cached_access_key_id().await);
-    set_var("AWS_SECRET_ACCESS_KEY", cached_secret_access_key().await);
-    if let Some(session_token) = cached_session_token().await {
-        set_var("AWS_SESSION_TOKEN", session_token);
-    }
-    clear_cached_credential().await;
-
-    let client = get_client().await;
-    let coll = client.database("aws").collection::<Document>("somecoll");
-    coll.find_one(doc! {}).await.unwrap();
-    assert!(cached_credential().await.is_none());
-
-    set_var("AWS_ACCESS_KEY_ID", "bad");
-    set_var("AWS_SECRET_ACCESS_KEY", "bad");
-    set_var("AWS_SESSION_TOKEN", "bad");
-
-    let client = get_client().await;
-    let coll = client.database("aws").collection::<Document>("somecoll");
-    match coll.find_one(doc! {}).await {
-        Ok(_) => panic!(
-            "find one should have failed with authentication error due to poisoned environment \
-             variables"
-        ),
-        Err(error) => assert!(error.is_auth_error()),
-    }
-
-    remove_var("AWS_ACCESS_KEY_ID");
-    remove_var("AWS_SECRET_ACCESS_KEY");
-    remove_var("AWS_SESSION_TOKEN");
-    clear_cached_credential().await;
-
-    let client = get_client().await;
-    let coll = client.database("aws").collection::<Document>("somecoll");
-    coll.find_one(doc! {}).await.unwrap();
-    assert!(cached_credential().await.is_some());
-
-    set_var("AWS_ACCESS_KEY_ID", "bad");
-    set_var("AWS_SECRET_ACCESS_KEY", "bad");
-    set_var("AWS_SESSION_TOKEN", "bad");
-
-    let client = get_client().await;
-    let coll = client.database("aws").collection::<Document>("somecoll");
-    coll.find_one(doc! {}).await.unwrap();
-
-    remove_var("AWS_ACCESS_KEY_ID");
-    remove_var("AWS_SECRET_ACCESS_KEY");
-    remove_var("AWS_SESSION_TOKEN");
-}
+// async fn get_client() -> Client {
+//     Client::with_uri_str(DEFAULT_URI.clone()).await.unwrap()
+// }
+
+// #[tokio::test]
+// async fn credential_caching() {
```

Comment:
Rather than commenting out these tests, we can add the [`#[ignore]`](https://doc.rust-lang.org/reference/attributes/testing.html#the-ignore-attribute) annotation to skip running them for now with a comment indicating that they'll be addressed in a future PR for RUST-1529